blueprints/controllers/ikea_e1743/ikea_e1743.yaml

```
@@ -261,9 +315,13 @@ condition:
         {%- endif -%}
         {%- endset -%}
         {{ trigger_action not in ["","None"] }}
+#**YAR** - ADD END
       # only for zigbee2mqtt, check if the event is relative to a real state change, and not only some minor changes in the sensor attributes
       # this is required since multiple state_changed events are fired for a single button press, with the result of the automation being triggered multiple times
-      - '{{ integration_id != "zigbee2mqtt" or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
+#**YAR** - Below needed to change if z2m entity or device not sure I got this right
+#**YAR**      - '{{ integration_id != "zigbee2mqtt" or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
+      - '{{ integration_id != "zigbee2mqtt" or trigger.payload != last_controller_event or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
```

```suggestion
      - '{{ not (integration_id == "zigbee2mqtt" and z2m_legacy) or trigger.payload != last_controller_event or trigger.event.data.new_state.state != trigger.event.data.old_state.state }}'
```

@Nicolai- equivalent to what you suggested, but that should make it clearer that the first two conditions are related.

---

Almost, I removed the `or trigger.payload != last_controller_event` part as this part was always true, when looking at my traces
I'm pretty sure its because `last_controller_event` is not defined in this scope, it's defined in the action scope, but from what I can read these variables are only available in that scope.

---

EDIT: Correct! Not sure how that ended up there, thanks for spotting that.